src/CallbackController.php

```diff
@@ -31,15 +36,31 @@ public function callback(Request $request)
         if (!$request->has('code')) {
             throw new AuthenticationException('No authorization code received');
         }
-        $token = $this->provider->getAccessToken('authorization_code', [
+        $token = $this->provider->getAccessToken(
+            'authorization_code',
+            [
             'code' => $request->get('code'),
-        ]);
+            ]
+        );
 
         $required = config('oidc-auth.required_claims');
+
+        // attempt to get serialized closure first
```

Comment:
we should probably do this only if not is_array, check if it is a string first, and drop the try-catch

Reply:
fixed in new revision

Reply:
> only if not is_array

like

```
if (is_array) {
} else {
  if (is_string) {
  }
  if (closure)  {
  }
}
```

it's less complicated this way imo